lib/crystalball/map_generator/described_class_strategy.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'crystalball/map_generator/base_strategy'
+require 'crystalball/map_generator/class_sources_detector'
+
+module Crystalball
+  class MapGenerator
+    # Map generator strategy to get paths to files contains definition of described_class and its
+    # ancestors.
+    class DescribedClassStrategy
+      include BaseStrategy
+      extend Forwardable
+
+      attr_reader :execution_detector
+
+      delegate %i[after_register before_finalize] => :execution_detector
+
+      def self.build(only: [], root: Dir.pwd)
+        hierarchy_fetcher = ClassSourcesDetector::HierarchyFetcher.new(only)
+        execution_detector = ClassSourcesDetector.new(root_path: root, hierarchy_fetcher: hierarchy_fetcher)
+
+        new(execution_detector: execution_detector)
+      end
+
+      def initialize(execution_detector:)
+        @execution_detector = execution_detector
+      end
+
+      def call(case_map, example)
+        yield case_map
+
+        described_class = example.metadata[:described_class]
+
+        case_map.push(*execution_detector.detect([described_class])) if described_class
```

Comment:
described_class is nil for string describes? E.g. `describe "mymodel" {}`

Reply:
yep, it will be nil